### PR TITLE
Build sample site using a docker image with hugo modules support

### PR DESCRIPTION
**Pull request by amuraru**
## Description
- made the main theme dir a go module
- use amuraru/hugo docker image that has hugo,git and golang
- define theme module in sample site as a hugo module instead of a go module
to always force a download/refresh

### CODE REVIEW

**Comment by amuraru** [comment]
@alexander-falca please review

File: publish-to-gh-pages.sh [hunk 1]
```diff
@@ -21,6 +21,13 @@ git clone --single-branch --branch gh-pages git@github.com:adobe/hugo-spectrum.g
 rm -rf site-pages/*
 
 hugo -v --source="$PWD/exampleSite" --destination="$PWD/site-pages" --ignoreCache
+docker run --rm \
```
Comment (alexander-falca) [inline publish-to-gh-pages.sh:23]: I think the previous line can be safely deleted.

**Review by alexander-falca — APPROVED** [review]
see minor comment and then :+1:

Thanks for building image with golang!

File: publish-to-gh-pages.sh (hunk 1 above)
Comment (amuraru) [reply publish-to-gh-pages.sh:23]: Good catch :)

**Review by alexander-falca — APPROVED** [review]
:+1:

**Pull request merged**